packages/gatsby/src/utils/jobs-manager.ts

```diff
@@ -24,13 +24,22 @@ const jobsInProcess = new Map()
 /** @type {Map<string, {job: InternalJob, deferred: pDefer.DeferredPromise<any>}>} */
 const externalJobsMap = new Map()
```

Comment:
can you pass the type to these two maps?  something like:

```ts
new Map<string, {job: InternalJob, pDefer.DeferedPromise<any>}>
```

Reply:
Yeah, I have some more changes locally, but I've been having some issues with converting the tests for it to typescript. Is looks like tests are failing due to the fact that the imports are different, which `jest.isolateModules` doesn't like. I'll work some more on those changes and push them :)